update: enhance zod schema validation for Login and Register pages

Maaayhan opened the pull request:
Resolved CP-126

[CP-126](https://beequant.atlassian.net/jira/software/projects/CP/boards/1?selectedIssue=CP-126)

---

## What happens here?
- update schema for the login page to ensure 'email' conforms to the standard and 'password' is not empty.
- update the schema for the registration page to ensure all fields conform to the standards and 'confirmPassword' matches 'password'.
- add multilingual support.
- add passwordSchema in 'utils' for reuse.

## How do I know this is working?
- when the user enters an invalid email address or leaves the password field empty on login page, a prompt message will appear.
<img width="538" alt="image" src="https://github.com/user-attachments/assets/9e271ec8-ce18-40f5-95e9-079bdef7a8a8">

- when the user inputs data that does not meet the standards on register page, a message will be displayed
- if the confirm password does not match the password, the message will be shown
<img width="537" alt="image" src="https://github.com/user-attachments/assets/7ca4d924-6a90-4a54-9444-4c24381d7a39">


## Any notes?

<!--
  New PR Checklist

- Have you tag the right reviewer?
- Have you dm the reviewer?
- Maybe @ the reviewer in channel just to be safe

- is CI passing in your pr?
- Have you update the ticket with the right status?
-->

NorrisWu0 commented on apps/client/src/module/auth/login-form.tsx:
```diff
@@ -15,12 +15,21 @@ import { SocialButton } from "../common/social-button";
+  email: z.string()
+    .min(1, { message: EmailErrorMsgs.Required })
+    .email({ message: EmailErrorMsgs.Invalid })
+    .regex(emailPattern, { message: EmailErrorMsgs.Invalid }),
```
What's special about this email pattern?

Maaayhan replied:
The email pattern is left when joi was used before, has the same effect as.email(). I'll delete that line

NorrisWu0 commented on apps/client/src/module/auth/register-form.tsx:
```diff
@@ -12,16 +12,51 @@ import { AuthFormContainer } from "./auth-form-container";
+    .regex(passwordPatten, { message: PasswordErrorMsgs.Invalid }),
   realName: z.string().optional(),
-  ref: z.string(),
+  ref: z.string().min(1, { message: RefErrorMsgs.Required }),
```
Oh. Do we have a default value for this?

Maaayhan replied:
yeah we have a pre-filled default value corresponding to the backend, I‘ll add a default value specification here

Maaayhan commented:
For the schema of displayName in register page, i found that if user input an empty string, the operation of converting to the default value "New User" was completed in the schema of the back-end, so I wrote the front-end part as: meet the format requirements of display or the length of string was zero.

NorrisWu0 commented on apps/client/src/module/auth/login-form.tsx:
```diff
@@ -17,23 +17,28 @@ import { login, LoginPayload } from "./auth-service";
+  const defaultValues = {
+    email: "",
+    password: "",
+    remeber: "",
```
oops, looks like a typo here

NorrisWu0 reviewed (dismissed):
lgtm, except that typo. feel free to merge this after that's fixed 🥳

Maaayhan committed:
update: enhance zod schema validation for Login and Register pages

- update schema for the login page to ensure 'email' conforms to the standard and 'password' is not empty.
- update the schema for the registration page to ensure all fields conform to the standards and 'confirmPassword' matches 'password'.
- add multilingual support
- add passwordSchema in 'utils' for reuse

Resolved CP-126
